vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -216,6 +216,11 @@ window.Vaadin.Flow.gridConnector = {
       if(!detailsVisibleOnClick) {
         return;
       }
+      // when grid is attached, newVal is not set and oldVal is undefined
+      // do nothing
+      if ((newVal == null) && (typeof oldVal === 'undefined')) {
```

Comment:
nit: oldVal is declared so no need to use typeof

Reply:
I removed typeof, Not sure if it's ok.

Reply:
Now it's comparing the value to a string 'undefined'. Should say `=== undefined`

Reply:
done